runtime/drivers/olap.go

```diff
@@ -31,7 +38,33 @@ type Statement struct {
 // Result wraps the results of query.
 type Result struct {
 	*sqlx.Rows
-	Schema *runtimev1.StructType
+	Schema    *runtimev1.StructType
+	cleanupFn func() error
+}
+
+// SetCleanupFunc sets a function, which will be called when the Result is closed.
+func (r *Result) SetCleanupFunc(fn func() error) {
+	if r.cleanupFn != nil {
+		panic("cleanup function already set")
+	}
+	r.cleanupFn = fn
+}
+
+// Close wraps rows.Close and calls the Result's cleanup function (if it is set).
+// Close should be idempotent.
+func (r *Result) Close() error {
```

Comment:
I see this method overrides the `Close()` method on `sqlx.Rows`, is it because in the `Result` struct `sqlx.Rows` parameter does not have a name ? this is new to me.

Reply:
Yep :)

Reply:
Ok I was looking for a similar thing but didn't knew about it